components/core/Profile.js

```diff
@@ -103,78 +101,495 @@ const STYLES_DESCRIPTION = css`
+  align-items: center;
+  font-size: ${Constants.typescale.lvl1};
+  cursor: pointer;
+  ${"" /* border: 1px solid ${Constants.system.lightBorder}; */}
+  border-radius: 4px;
+  margin-bottom: 8px;
+  background-color: ${Constants.system.white};
+`;
+
+const STYLES_USER = css`
+  display: grid;
+  grid-template-columns: auto 1fr;
+  align-items: center;
+  margin: 16px;
+  color: ${Constants.system.brand};
+  font-family: ${Constants.font.medium};
+  font-size: ${Constants.typescale.lvl1};
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    margin: 12px 16px;
+  }
+`;
+
+const STYLES_DIRECTORY_PROFILE_IMAGE = css`
+  background-color: ${Constants.system.foreground};
+  background-size: cover;
+  background-position: 50% 50%;
+  height: 24px;
+  width: 24px;
+  margin-right: 16px;
+  border-radius: 4px;
+`;
+
+const STYLES_MESSAGE = css`
+  color: ${Constants.system.black};
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+
+  @media (max-width: 1000px) {
+    display: none;
+  }
+`;
+
+const STYLES_DIRECTORY_NAME = css`
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+`;
+
+const STYLES_COPY_INPUT = css`
+  pointer-events: none;
+  position: absolute;
+  opacity: 0;
+`;
+
+function UserEntry({ user, button, onClick, message, external, url }) {
+  return (
+    <div key={user.username} css={STYLES_USER_ENTRY}>
+      {external ? (
+        <a css={STYLES_USER} style={{ textDecoration: "none" }} href={url}>
+          <div
+            css={STYLES_DIRECTORY_PROFILE_IMAGE}
+            style={{ backgroundImage: `url(${user.data.photo})` }}
+          />
+          <span css={STYLES_DIRECTORY_NAME}>
+            {user.data.name || `@${user.username}`}
+            {message ? <span css={STYLES_MESSAGE}>{message}</span> : null}
+          </span>
+        </a>
+      ) : (
+        <div css={STYLES_USER} onClick={onClick}>
+          <div
+            css={STYLES_DIRECTORY_PROFILE_IMAGE}
+            style={{ backgroundImage: `url(${user.data.photo})` }}
+          />
+          <span css={STYLES_DIRECTORY_NAME}>
+            {user.data.name || `@${user.username}`}
+            {message ? <span css={STYLES_MESSAGE}>{message}</span> : null}
+          </span>
+        </div>
+      )}
+      {button}
+    </div>
+  );
+}
+
 export default class Profile extends React.Component {
+  _ref = null;
+  lastLength = null;
+
   state = {
-    exploreSlates: [],
+    tab: 1,
+    view: 0,
+    fileTab: 0,
+    slateTab: 0,
+    peerTab: 0,
+    copyValue: "",
+    contextMenu: null,
+    publicSlates: [],
+    publicFiles: [],
+    pseudoPrivateFiles: [],
+    isFollowing: this.props.external
+      ? false
+      : !!this.props.viewer.subscriptions.filter((entry) => {
+          return entry.target_user_id === this.props.data.id;
+        }).length,
+  };
+
+  componentDidMount = async () => {
+    await this.filterByVisibility();
+  };
+
+  componentDidUpdate = async () => {
```

Comment:
Currently, if you go from one profile to another, it'll stay on the same tab b/c react perceives it as the same component (doesn't restart state or call component did mount). So to manually take the user to the default slates tab, I'm adding this
```
  componentDidUpdate = async (prevProps) => {
    if (this.props.creator?.id !== prevProps.creator?.id) {
      this.setState({ tab: 1, view: 0, fileTab: 0, slateTab: 0, peerTab: 0 });
    }
```

Reply:
hmmm, good catch! not sure if prevProps work tho. remember the bug we talked about?

Reply:
i tested it out and it works fine for this case